controllers/machinedeployment_ondelete.go

```diff
@@ -0,0 +1,182 @@
+/*
```

Comment:
Could we name this file `machinedeployment_rollout_ondelete`?

Reply:
we can! and done